node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminStateUpdater.java

```diff
@@ -64,6 +75,18 @@ public NodeAdminStateUpdater(
         this.orchestrator = orchestrator;
         this.dockerHostHostName = dockerHostHostName;
         this.lastTick = clock.instant();
+
+        storageMaintainer.ifPresent(maintainer -> specVerifierScheduler.scheduleWithFixedDelay(() -> {
+            if (currentState != RESUMED) return;
+
+            try {
+                String hardwareDivergence = maintainer.getHardwardDivergence();
+                NodeAttributes nodeAttributes = new NodeAttributes().withHardwareDivergence(hardwareDivergence);
+                nodeRepository.updateNodeAttributes(dockerHostHostName, nodeAttributes);
+            } catch (RuntimeException e) {
+                log.log(Level.WARNING, "Failed to report hardware divergence", e);
```

Comment:
OK, so not too bad then... consider adding one top-level ("number of times something failed for spec").